### DISCUSSION TITLE
Allow npm build to manipulate shipped filesystem (support Critical CSS, etc)

### DISCUSSION THREAD

**Discussion by timkelty** [craftcms/cloud#21] [Feature Requests & Enhancements]
Currently, the NPM build exclusively ships artifacts to the CDN.
This means if the build process writes things outside of the `webroot`, it will be ignored, as the resulting files are not included back into the shipped filesystem.

Use-cases include:
- Critical CSS html templates
- Fractal (or other component library) builds (twig templates)
- Front-end build tools that write Twig templates, eg https://github.com/brianjhanson/vite-plugin-craftcms

/cc @shennyg

**Comment by angrybrad** [comment]
This has been implemented and deployed for all projects.
